### PR TITLE
feat: Hack#3 - Count and display the amount of items in a Webflow Col…

### PR DESCRIPTION
**Pull request by aexshafii**
…lection List

### CODE REVIEW

File: webflow-hacks/cms-hacks/3/3.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+document.addEventListener('DOMContentLoaded', () => {
+  // get the amount (length) of ".cms-job-counter" classes
+  // and assign that amount to a constant named jobsCounter
+  const jobsCounter = document.querySelectorAll('.cms-jobs-counter').length;
+
+  // select the element with class ".hack3-jobs-amount"
+  const jobsCount: HTMLElement = document.querySelector('.hack3-jobs-amount') as HTMLElement;
```
Comment (alexiglesias93) [inline webflow-hacks/cms-hacks/3/3.ts]: @aexshafii I would add null checks here to make this script safe.
This is the primary reason why we use Typescript 💪🏻 

```typescript
const jobsCount = document.querySelector<HTMLElement>('.hack3-jobs-amount');
if (!jobsCount) return;
```

Comment (aexshafii) [reply webflow-hacks/cms-hacks/3/3.ts]: I see. Making the change now. Thanks!

**Review by alexiglesias93 — APPROVED** [review]

File: webflow-hacks/cms-hacks/3/3.ts [hunk 2]
```diff
@@ -1,11 +1,12 @@
 document.addEventListener('DOMContentLoaded', () => {
-  // get the amount (length) of ".cms-job-counter" classes
+  // get the amount (length) of elements with data attribute[fs-hacks-element="CMS_JOBS_COUNTER"]
   // and assign that amount to a constant named jobsCounter
-  const jobsCounter = document.querySelectorAll('.cms-jobs-counter').length;
+  const jobsCounter = document.querySelectorAll('[fs-hacks-element="CMS_JOBS_COUNTER"]').length;
```
Comment (alexiglesias93) [inline webflow-hacks/cms-hacks/3/3.ts]: Naming convention for attributes is all kebab-case:
`[fs-hacks-element="cms-jobs-counter"]`

File: webflow-hacks/cms-hacks/3/3.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+document.addEventListener('DOMContentLoaded', function () {
+  // get the amount (length) of ".cms-job-counter" classes
+  // and assign that amount to a constant named jobsCounter
+  var jobsCounter = document.querySelectorAll('[fs-hacks-element="CMS_JOBS_COUNTER"]').length;
+  console.log(jobsCounter);
```
Comment (blessochampion) [inline webflow-hacks/cms-hacks/3/3.js]: I think we should remove the log here.

Comment (aexshafii) [reply webflow-hacks/cms-hacks/3/3.js]: should be gone already! this was from a previous commit

**Comment by aexshafii** [comment]
ready for review

File: webflow-hacks/cms-hacks/3/3.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  const JOBS_COUNTER_SELECTOR = '[fs-hacks-element="cms-jobs-counter"]';
+  const JOB_COUNT_SELECTOR = '[fs-hacks-element="hack3-job-count"]';
+
+  // get the amount (length) of elements with selector JOBS_COUNTER_SELECTOR
+  // and assign that amount to a constant named jobsCounter
+  const jobsCounter = document.querySelectorAll(JOBS_COUNTER_SELECTOR).length;
+
+  // select the element with data attribute JOB_COUNT_SELECTOR
+  const jobCount = document.querySelector<HTMLElement>(JOB_COUNT_SELECTOR);
+  if (!jobCount) return;
```
Comment (alexiglesias93) [inline webflow-hacks/cms-hacks/3/3.ts]: I would swap the order of declaring `jobsCounter` and `jobCount`.

If `jobCount` does not exist on the page, the function returns. So there's no need to query the `jobsCounter` elements.

**Comment by aexshafii** [comment]
ready for review v2

**Review by alexiglesias93 — APPROVED** [review]

**Pull request merged**